oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateTest.java

```diff
@@ -75,10 +82,97 @@ public void multiValuedBinarySize() throws Exception {
+        assertEquals(13, p.count());
+
+        assertNotNull(((DocumentPropertyState) p).getCompressedValue());
+
         reads.clear();
         assertEquals(BLOB_SIZE, p.size(0));
         // must not read the blob via stream
         assertEquals(0, reads.size());
     }
 
+    @Test
+    public void stringBelowThresholdSize() throws Exception {
+        NodeBuilder builder = ns.getRoot().builder();
+        builder.child(TEST_NODE).setProperty("p", "dummy", Type.STRING);
+        TestUtils.merge(ns, builder);
+
+        PropertyState p = ns.getRoot().getChildNode(TEST_NODE).getProperty("p");
+        assertEquals(Type.STRING, Objects.requireNonNull(p).getType());
+        assertEquals(1, p.count());
+
+        assertNull(((DocumentPropertyState) p).getCompressedValue());
+
+        reads.clear();
+        assertEquals(5, p.size(0));
+        // must not read the string via stream
+        assertEquals(0, reads.size());
+    }
+
+    @Test
+    public void stringAboveThresholdSize() throws Exception {
+        NodeBuilder builder = ns.getRoot().builder();
+        builder.child(TEST_NODE).setProperty("p", STRING_HUGEVALUE, Type.STRING);
+        TestUtils.merge(ns, builder);
+
+        PropertyState p = ns.getRoot().getChildNode(TEST_NODE).getProperty("p");
+        assertEquals(Type.STRING, Objects.requireNonNull(p).getType());
+        assertEquals(1, p.count());
+
+        assertNotNull(((DocumentPropertyState) p).getCompressedValue());
+
+        reads.clear();
+        assertEquals(1050, p.size(0));
+        // must not read the string via streams
+        assertEquals(0, reads.size());
+    }
+
+    @Test
+    public void compressValueThrowsException() throws CommitFailedException, IOException {
+        DocumentNodeStore mockDocumentStore = mock(DocumentNodeStore.class);
+        Compression mockCompression = mock(Compression.class);
+        when(mockCompression.getOutputStream(any(OutputStream.class))).thenThrow(new IOException("Compression failed"));
+
+        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", STRING_HUGEVALUE, mockCompression);
+
+        verify(mockCompression, times(1)).getOutputStream(any(OutputStream.class));
+
+    }
+
+    @Test
+    public void uncompressValueThrowsException() throws CommitFailedException, IOException {
+
+        DocumentNodeStore mockDocumentStore = mock(DocumentNodeStore.class);
+        Compression mockCompression = mock(Compression.class);
+        OutputStream mockOutputStream= mock(OutputStream.class);
+        when(mockCompression.getOutputStream(any(OutputStream.class))).thenReturn(mockOutputStream);
+        when(mockCompression.getInputStream(any(InputStream.class))).thenThrow(new IOException("Compression failed"));
+
+        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", STRING_HUGEVALUE, mockCompression);
+        documentPropertyState.getValue(Type.STRING);
```

Comment:
could you wrap this in a `assertEquals()` or similar?

Reply:
Done